nano/secure/rep_weights.cpp

```diff
@@ -0,0 +1,139 @@
+{
+	if (rep_1 != rep_2)
+	{
+		auto previous_weight_1{ rep_weight_store.get (txn_a, rep_1) };
+		auto previous_weight_2{ rep_weight_store.get (txn_a, rep_2) };
+		auto new_weight_1 = previous_weight_1 + amount_1;
+		auto new_weight_2 = previous_weight_2 + amount_2;
+		put_store (txn_a, rep_1, previous_weight_1, new_weight_1);
+		put_store (txn_a, rep_2, previous_weight_2, new_weight_2);
+		nano::lock_guard<nano::mutex> guard (mutex);
+		put_cache (rep_1, new_weight_1);
+		put_cache (rep_2, new_weight_2);
+	}
+	else
+	{
+		representation_add (txn_a, rep_1, amount_1 + amount_2);
+	}
+}
+
+void nano::rep_weights::representation_put (nano::account const & account_a, nano::uint128_t const & representation_a)
+{
+	nano::lock_guard<nano::mutex> guard (mutex);
+	put_cache (account_a, representation_a);
+}
+
+nano::uint128_t nano::rep_weights::representation_get (nano::account const & account_a) const
+{
+	nano::lock_guard<nano::mutex> lk (mutex);
+	return get (account_a);
+}
+
+/** Makes a copy */
+std::unordered_map<nano::account, nano::uint128_t> nano::rep_weights::get_rep_amounts () const
+{
+	nano::lock_guard<nano::mutex> guard (mutex);
+	return rep_amounts;
+}
+
+void nano::rep_weights::copy_from (nano::rep_weights & other_a)
+{
+	nano::lock_guard<nano::mutex> guard_this (mutex);
+	nano::lock_guard<nano::mutex> guard_other (other_a.mutex);
+	for (auto const & entry : other_a.rep_amounts)
+	{
+		auto prev_amount (get (entry.first));
+		put_cache (entry.first, prev_amount + entry.second);
+	}
+}
+
+void nano::rep_weights::put_cache (nano::account const & account_a, nano::uint128_union const & representation_a)
+{
+	auto it = rep_amounts.find (account_a);
+	if (representation_a.is_zero ())
+	{
+		if (it != rep_amounts.end ())
+		{
+			rep_amounts.erase (it);
+		}
+	}
+	else
+	{
+		auto amount = representation_a.number ();
+		if (it != rep_amounts.end ())
+		{
+			it->second = amount;
+		}
+		else
+		{
+			rep_amounts.emplace (account_a, amount);
+		}
+	}
+}
+
+void nano::rep_weights::put_store (store::write_transaction const & txn_a, nano::account const & rep_a, nano::uint128_t const & previous_weight_a, nano::uint128_t const & new_weight_a)
+{
+	if (new_weight_a.is_zero ())
+	{
+		if (!previous_weight_a.is_zero ())
+		{
+			rep_weight_store.del (txn_a, rep_a);
+		}
+	}
+	else
+	{
+		rep_weight_store.put (txn_a, rep_a, new_weight_a);
+	}
+}
+
+nano::uint128_t nano::rep_weights::get (nano::account const & account_a) const
+{
+	auto it = rep_amounts.find (account_a);
+	if (it != rep_amounts.end ())
+	{
+		return it->second;
+	}
+	else
+	{
+		return nano::uint128_t{ 0 };
+	}
+}
+
+std::size_t nano::rep_weights::size () const
+{
+	nano::lock_guard<nano::mutex> guard (mutex);
+	return rep_amounts.size ();
+}
+
+std::unique_ptr<nano::container_info_component> nano::collect_container_info (nano::rep_weights const & rep_weights, std::string const & name)
```

Comment:
We've been changing these friend + free functions that collect weight to instead be member functions.

Reply:
Converted it into a member function